lib/tokenize.cpp

```diff
@@ -8401,6 +8401,9 @@ void Tokenizer::findGarbageCode() const
         else if (Token::Match(tok, "%assign% [") && Token::simpleMatch(tok->linkAt(1), "] ;"))
             syntaxError(tok, tok->str() + "[...];");
 
+        else if (Token::Match(tok, "[({<] %assign%"))
```

Comment:
isn't this executed before the symboldatabase? I wonder if the changes in symboldatabase+testgarbage is technically separated?

Reply:
Yes, those are separate changes. `validateVariables()` was (a) deactivated and (b) not doing sufficient checking to prevent the crash, so I added the `syntaxError`.